libs/cardano-ledger-pretty/src/Cardano/Ledger/Pretty/Mary.hs

```diff
@@ -17,24 +17,31 @@ import Cardano.Ledger.ShelleyMA.Timelocks
 import Cardano.Ledger.ShelleyMA.TxBody
 import Prettyprinter (hsep, viaShow)
 
-ppValue :: MaryValue crypto -> PDoc
-ppValue v = case gettriples' v of
-  (n, triples, []) -> ppSexp "Value" [ppCoin (Coin n), ppList pptriple triples]
-  (n, triples, bad) -> ppSexp "Value" [ppCoin (Coin n), ppList pptriple triples, ppString "Bad " <> ppList ppPolicyID bad]
-  where
-    pptriple (i, asset, num) = hsep [ppPolicyID i, ppAssetName asset, ppInteger num]
-
 ppPolicyID :: PolicyID crypto -> PDoc
 ppPolicyID (PolicyID sh) = ppScriptHash sh
 
+instance PrettyA (PolicyID crypto) where prettyA x = ppSexp "PolicyID" [ppPolicyID x]
+
 ppAssetName :: AssetName -> PDoc
 ppAssetName = viaShow
 
-instance PrettyA (MaryValue crypto) where prettyA = ppValue
+instance PrettyA AssetName where prettyA x = ppSexp "AssetName" [ppAssetName x]
 
-instance PrettyA (PolicyID crypto) where prettyA x = ppSexp "PolicyID" [ppPolicyID x]
+ppMultiAsset :: MultiAsset crypto -> PDoc
+ppMultiAsset m = ppList pptriple (flattenMultiAssets m)
+  where
+    pptriple (i, asset, num) = hsep [ppPolicyID i, ppAssetName asset, ppInteger num]
 
-instance PrettyA AssetName where prettyA x = ppSexp "AssetName" [ppAssetName x]
+instance CC.Crypto crypto => PrettyA (MultiAsset crypto) where prettyA x = ppSexp "MultiAsset" [ppMultiAsset x]
+
+ppValue :: MaryValue crypto -> PDoc
+ppValue v = case gettriples' v of
+  (n, triples, []) -> ppSexp "Value" [ppCoin (Coin n), ppList pptriple triples]
+  (n, triples, bad) -> ppSexp "Value" [ppCoin (Coin n), ppList pptriple triples, ppString "Bad " <> ppList ppPolicyID bad]
+  where
+    pptriple (i, asset, num) = hsep [ppPolicyID i, ppAssetName asset, ppInteger num]
```

Comment:
We can simplify this. See my comment on `gettripples'` in [eras/shelley-ma/impl/src/Cardano/Ledger/Mary/Value.hs](https://github.com/input-output-hk/cardano-ledger/pull/2954/files#diff-ec516a1c5b807ba71f5b0a126cd17bf9d406d6d91e48ebf056ff80e5b73634ed)
```suggestion
ppValue (MaryValue n m) = ppSexp "Value" $ [ppCoin (Coin n), ppMultiAsset m] ++ ppBad 
  where
    ppBad = case getBadMultiAsset m of
       [] -> []
       bad -> [ppString "Bad " <> ppList ppPolicyID bad]
```